app/Http/Controllers/Agreement/AgreementController.php

```diff
@@ -95,10 +95,13 @@ public function update(AgreementRequest $request)
     {
         $inputs = $request->all();
 
-        \DB::beginTransaction();
-        try {
-            $agreement = Agreement::find($inputs['id']);
-            $agreement->fill([
+        // 編集権限がなければリダイレクト
+        if(Auth::user()->name === $inputs['client_name'] || Auth::user()->admin_status === 1){
```

Comment:
```suggestion
        if(Auth::user()->name !== $inputs['client_name'] || Auth::user()->admin_status !== 1){
          return redirect(route('agreement.index'))->with('flash_message', '操作権限がありません');
         }
```
C言語とかの名残ではあるんですがプログラムは上から順番に処理が実行されていくので
終わらせられるのであれば処理の最初にreturnしっちゃった方が良いです。